api/controllers/web/wraps.py

```diff
@@ -54,25 +55,31 @@ def decode_jwt_token():
         if not end_user:
             raise NotFound()
 
-        _validate_web_sso_token(decoded, system_features)
+        _validate_web_sso_token(decoded, system_features, app_code)
 
         return app_model, end_user
     except Unauthorized as e:
         if system_features.sso_enforced_for_web:
-            raise WebSSOAuthRequiredError()
+            app_web_sso_enabled = EnterpriseService.get_app_web_sso_enabled(app_code).get('enabled', False)
+            if app_web_sso_enabled:
+                raise WebSSOAuthRequiredError()
 
         raise Unauthorized(e.description)
 
 
-def _validate_web_sso_token(decoded, system_features):
+def _validate_web_sso_token(decoded, system_features, app_code):
+    app_web_sso_enabled = False
```

Comment:
This is unnecessary is you declared as `EnterpriseService.get_app_web_sso_enabled(app_code).get('enabled', False)`